rest-server/src/components/users/userControllers.js

```diff
@@ -0,0 +1,65 @@
+import { addUserHelper, fetchAllUserHelper, fetchSingleUserHelper, updateUserHelper, deleteUserHelper } from './userHelpers';
+
+const addUser = async (req, res) => {
+  try {
+    const newUser = await addUserHelper(req.body);
+    res.status(201).send(newUser);
+  }
+  catch (err) {
+    res.status(400).send({error: 'Could not create User'});
+  }
+};
+
+const fetchAllUser = async (req, res) => {
+  try {
+    const allUser = await fetchAllUserHelper();
+    res.status(200).send(allUser);
+  }
+  catch (err) {
+    res.status(400).send({error: 'Could not find Users'});
+  }
+};
+
+const fetchSingleUser = async (req, res) => {
+  try{
+    const singleUser = await fetchSingleUserHelper(req.params);
+    res.status(200).send(singleUser);
+  }
+  catch (err) {
+    res.status(400).send({error: 'Could not find User'});
+  }
+};
+
+const updateUser = async (req, res) => {
```

Comment:
This method is a bit confusing. It looks like you are trying to combine PUT (or POST) and a PATCH. Meaning, you are trying to add a user as well as update a user based on fragmented data and not a complete object.

This could probably use some additional thought and abstraction into separate handlers.

Reply:
You are correct, I got ahead of myself there and overthought the PUT handler. I have made the adjustments to separate PUT/PATCH and will push my changes soon.